jax/_src/interpreters/pxla.py

```diff
@@ -1867,8 +1867,9 @@ def _raise_warnings_or_errors_for_jit_of_pmap(
          "jit-of-pmap can lead to inefficient data movement, as the outer jit "
          "does not preserve sharded data representations and instead collects "
          "input and output arrays onto a single device. "
-         "Consider removing the outer jit unless you know what you're doing. "
-         "See https://github.com/jax-ml/jax/issues/2926.")
+         "Use jax.experimental.shard_map instead of pmap under jit compilation. "
```

Comment:
Let's suggest both!

Remove `jit` or use `shard_map`. And remove the `file issue` sentence.

Reply:
Thanks! Updated the warning message accordingly.